qt-bugs@ issue : none yet
bugs.kde.org number : none
applied: no
author: Dirk Mueller <mueller@kde.org>


Index: src/kernel/qasyncimageio.h
===================================================================
--- src/kernel/qasyncimageio.h	(revision 467438)
+++ src/kernel/qasyncimageio.h	(working copy)
@@ -44,6 +44,9 @@
 
 #ifndef QT_NO_ASYNC_IMAGE_IO
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 class Q_EXPORT QImageConsumer {
 public:
Index: src/kernel/qucomextra_p.h
===================================================================
--- src/kernel/qucomextra_p.h	(revision 467438)
+++ src/kernel/qucomextra_p.h	(working copy)
@@ -54,6 +54,9 @@
 #include <private/qucom_p.h>
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 class QVariant;
 
Index: src/kernel/qnetworkprotocol.h
===================================================================
--- src/kernel/qnetworkprotocol.h	(revision 467438)
+++ src/kernel/qnetworkprotocol.h	(working copy)
@@ -47,6 +47,10 @@
 
 #ifndef QT_NO_NETWORKPROTOCOL
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 class QNetworkProtocol;
 class QNetworkOperation;
 class QTimer;
Index: src/kernel/qimageformatinterface_p.h
===================================================================
--- src/kernel/qimageformatinterface_p.h	(revision 467438)
+++ src/kernel/qimageformatinterface_p.h	(working copy)
@@ -40,6 +40,9 @@
 #include <private/qcom_p.h>
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 //
 //  W A R N I N G
Index: src/dialogs/qfiledialog.h
===================================================================
--- src/dialogs/qfiledialog.h	(revision 467438)
+++ src/dialogs/qfiledialog.h	(working copy)
@@ -58,6 +58,10 @@
 #include "qurlinfo.h"
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 #ifndef QT_NO_FILEDIALOG
 
 class Q_EXPORT QFileIconProvider : public QObject
Index: src/tools/qucom_p.h
===================================================================
--- src/tools/qucom_p.h	(revision 467438)
+++ src/tools/qucom_p.h	(working copy)
@@ -59,6 +59,10 @@
 #undef check
 #endif
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 struct QUObject;
 struct QUInterfaceDescription;
 struct QUnknownInterface;
Index: src/tools/qcom_p.h
===================================================================
--- src/tools/qcom_p.h	(revision 467438)
+++ src/tools/qcom_p.h	(working copy)
@@ -53,6 +53,10 @@
 #include "quuid.h"
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 #ifndef QT_NO_COMPONENT
 
 class QObject;
Index: src/widgets/qtooltip.h
===================================================================
--- src/widgets/qtooltip.h	(revision 467438)
+++ src/widgets/qtooltip.h	(working copy)
@@ -42,6 +42,9 @@
 
 #ifndef QT_NO_TOOLTIP
 
+#if __GNUC__  - 0 > 3
+#pragma GCC system_header
+#endif
 
 class QTipManager;
 class QIconViewToolTip;
Index: src/widgets/qmainwindow.cpp
===================================================================
--- src/widgets/qmainwindow.cpp	(revision 467438)
+++ src/widgets/qmainwindow.cpp	(working copy)
@@ -287,8 +287,9 @@
 {
 public:
     QHideToolTip( QWidget *parent ) : QToolTip( parent ) {}
+    ~QHideToolTip() {}
 
-    void maybeTip( const QPoint &pos );
+    virtual void maybeTip( const QPoint &pos );
 };
 #endif
 
